src/app/products/charges/create-charge/create-charge.component.html

```diff
@@ -0,0 +1,200 @@
+              <mat-label>Charge Frequency</mat-label>
+              <mat-select required formControlName="feeFrequency">
+                <mat-option *ngFor="let feeFrequency of chargesTemplateData.feeFrequencyOptions" [value]="feeFrequency.id">
+                  {{ feeFrequency.value }}
+                </mat-option>
+              </mat-select>
+              <mat-error *ngIf="chargeForm.controls.feeFrequency.hasError('required')">
+                Charge Frequency is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field *ngIf="chargeForm.controls.chargeTimeType.value === 9  && chargeForm.controls.addFeeFrequency.value" fxFlex="48%">
+              <mat-label>Frequency Interval</mat-label>
+              <input type="number" step="1" matInput required formControlName="feeInterval">
+              <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('required') && !chargeForm.controls.feeInterval.hasError('pattern')">
+                Frequency Interval is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('pattern')">
+                Frequency Interval <strong>must be a positive integer</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field *ngIf="chargeForm.controls.chargeTimeType.value === 6 || chargeForm.controls.chargeTimeType.value === 7" fxFlex="48%">
+              <mat-label>Due Date</mat-label>
+              <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="dueDatePicker" required formControlName="feeOnMonthDay">
+              <mat-datepicker-toggle matSuffix [for]="dueDatePicker"></mat-datepicker-toggle>
+              <mat-datepicker #dueDatePicker></mat-datepicker>
+              <mat-error *ngIf="chargeForm.controls.feeOnMonthDay.hasError('required')">
+                Due Date is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="48%" *ngIf="chargeForm.controls.chargeTimeType.value === 7 || chargeForm.controls.chargeTimeType.value === 11">
+              <mat-label>Repeat Every ({{ repeatEveryLabel }})</mat-label>
+              <input type="number" step="1" matInput required formControlName="feeInterval">
+              <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('required') && !chargeForm.controls.feeInterval.hasError('pattern')">
+                Repeat Every ({{ repeatEveryLabel }}) is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="(chargeForm.controls.feeInterval.hasError('min') || chargeForm.controls.feeInterval.hasError('max')) && !chargeForm.controls.feeInterval.hasError('pattern')">
+                Repeat Every (Months) <strong>must be between 1 and 12</strong>
```

Comment:
I just wanted you to avoid hardcoding of Months in HTML. Won't it mean the same if we use the variable?

Reply:
It will mean the same if we use the variable, I've made the changes.